astropy/time/core.py

```diff
@@ -133,6 +135,12 @@ class TimeInfo(MixinInfo):
     @property
     def _represent_as_dict_attrs(self):
         method = self.serialize_method[self._serialize_context]
+
+        # These two formats cannot be serialized with formatted_value, so always
+        # use jd1_jd2 instead.
+        if self._parent.format in ('datetime', 'ymdhms'):
```

Comment:
It could also be done right now with the existing mechanism for splitting into different attributes like `sc.ra` and `sc.dec`.  Would there be benefit to using structured columns instead?

Overall, if we are trying to be principled about all this, it suggests that we might want to move the represent and construct code for special-case formats into the format classes themselves.